cognite/neat/rules/_analysis/_information_rules.py

```diff
@@ -2,28 +2,78 @@
 import logging
 import warnings
 from collections import defaultdict
-from typing import Any
+from typing import Any, cast
 
 import pandas as pd
 from pydantic import ValidationError
 
-from cognite.neat.rules.models._rules._types import ClassEntity, EntityTypes, ParentClassEntity
+from cognite.neat.rules.models._rules._types._base import ClassEntity, EntityTypes, ParentClassEntity, ReferenceEntity
 from cognite.neat.rules.models._rules.base import SchemaCompleteness
 from cognite.neat.rules.models._rules.information_rules import InformationClass, InformationProperty, InformationRules
 from cognite.neat.rules.models.rdfpath import TransformationRuleType
 from cognite.neat.utils.utils import get_inheritance_path
 
-from ._base import BaseAnalysis
+from ._base import BaseAnalysis, DataModelingScenario
 
 
 class InformationArchitectRulesAnalysis(BaseAnalysis):
     def __init__(self, rules: InformationRules):
         self.rules = rules
 
-    def class_parent_pairs(self) -> dict[ClassEntity, list[ParentClassEntity]]:
+    @property
+    def data_modeling_scenario(self) -> DataModelingScenario:
+        if not self.rules.reference:
+            return DataModelingScenario.from_scratch
+
+        if self.rules.metadata.namespace == self.rules.reference.metadata.namespace:
+            return DataModelingScenario.extend_reference
+        else:
+            return DataModelingScenario.reuse_components
```

Comment:
Reuse here is the same as a solution model?

Reply:
I will rename it to `solution_model_creation`